### DISCUSSION TITLE
好奇一下在Vercel下部署时AccessToken是如何刷新的？

### DISCUSSION THREAD

**Discussion by aidenlx** [spencerwooo/onedrive-vercel-index#114] [Q&A] [answered]
我看了下源码，看上去并没有从process.env.ACCESS_TOKEN读取值，有点好奇两个问题：
1. 环境变量设置的ACCESS_TOKEN具体是在哪里有用到呢？
2. 在vercel下通过getAccessToken方法获得的AccessToken过期了之后是怎么刷新的？我部署在自己的服务器时会遇到过一段时间返回500（后台请求返回401），我自己写了个遇到401尝试重新获取AccessToken的函数，好奇在平台部署是怎么解决的？

**Comment by aidenlx** [comment]
顶一下

**Reply by York618** [reply]
Serverless，你可以看作是一个群租服务器，写好代码放上面就行

**Reply by aidenlx** [reply]
我知道Serverless的话是偏函数式，有请求才激活的，难不成是平台定时重新部署服务进而更新AccessToken吗？

**Comment by spencerwooo** [comment]
这个确实是个奇妙的问题，一部分刷新的代码继承自 [最最最初的 Cloudflare Worker 版本](https://github.com/heymind/OneDrive-Index-Cloudflare-Worker)，甚至不是我维护的代码，目前移植到 Vercel 上确实没有什么变化，能自动刷新，不过内在原理我可能需要点时间才能排查明了了。

**Reply by aozaki-kuro** [reply]
https://github.com/WangNingkai/OLAINDEX/
我想起来还有个不错的1drv index项目，不过是php写的，token刷新部分不知道能不能作为参考？

**Comment by aozaki-kuro** [comment]
希望是可以有办法部署到自己服务器上，毕竟自己服务器的CPU时间比免费版长得多，总体运行性能可能会更好🤔

**Comment by myl7** [comment]
https://github.com/spencerwooo/onedrive-vercel-index/blob/7e22e2dd8eca2012d0bd426ecea12f8f7a282722/pages/api/index.ts#L19-L44

不知道找得对不对

**Comment by spencerwooo** [comment, marked as answer]
新版本的 token 刷新可以解释了~

**Reply by Nesive** [reply]
我的也是redis ttl过期后就变为空，这里的新版本指的是哪个版本呢？
